clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -1385,6 +1385,12 @@ void EmitAssemblyHelper::RunOptimizationPipeline(
     // configure the pipeline.
     OptimizationLevel Level = mapToLevel(CodeGenOpts);
 
+    if (LangOpts.SYCLIsDevice)
```

Comment:
ESIMD verification should ideally run even if LLVM passes are disabled, as this is logically part of FE semantic analysis.
@sndmitriev - what do you think?

Reply:
> ESIMD verification should ideally run even if LLVM passes are disabled, as this is logically part of FE semantic analysis.
@sndmitriev - what do you think?

Please note that `disable-llvm-passes` disables even such passes as AlwaysInline, which are required to preserve the semantics of `always_inline` attribute, i.e. the flag literally says disable `llvm` passes, not optimization or transformation passes, but `llvm` passes.

My understanding is that the flag is not intended to be used by end users unless they know what they are doing and therefore, I would prefer to leave ESIMD verification disabled if `-disable-llvm-passes` is present

Reply:
I agree with Alexey. `-disable-llvm-passes` is supposed to disable all LLVM passes with no exceptions.
I think what Konst has in mind is that ESIMD verifier pass should run even with `-O0`.

Reply:
> ESIMD verification should ideally run even if LLVM passes are disabled, as this is logically part of FE semantic analysis.

Can we move the logic of this pass to FE Sema then?

Reply:
@kbobrovs @sndmitriev Could you please provide any comments?

Reply:
Sorry for delay.
Unfortunately, we can't move this to Sema as it was identified as very time-consuming task by the FE team, so we went with the simpler solution.

> I think what Konst has in mind is that ESIMD verifier pass should run even with -O0.

Yes, exactly.